src/test/java/org/carlspring/cloud/storage/s3fs/path/S3PathTest.java

```diff
@@ -125,17 +125,6 @@ void relativize()
     }
 
     // register
-    @Test
-    void registerWithEventsThrowException()
-    {
-        // We're expecting an exception here to be thrown
-        Exception exception = assertThrows(UnsupportedOperationException.class, () -> {
-            forPath("file1").register(null);
-        });
-
-        assertNotNull(exception);
-    }
-
     @Test
     void registerThrowException()
```

Comment:
This method could be renamed to `registerWatcherShouldThrowException` which would make it clear what's the intended result.

```suggestion
    /**
     * This covers {@link S3Path#register (WatchService watcher, WatchEvent.Kind<?>... events))
     */
    @Test
    void registerWatcherShouldThrowUnsupportedOperationException()
```